stacks-node/src/burnchains/rpc/rpc_transport/mod.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright (C) 2025 Stacks Open Internet Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A simple JSON-RPC transport client using [`StacksHttpRequest`] for HTTP communication.
+//!
+//! This module provides a wrapper around basic JSON-RPC interactions with support
+//! for configurable authentication and timeouts. It serializes requests and parses
+//! responses while exposing error types for network, parsing, and service-level issues.
+
+use std::io;
+use std::time::Duration;
+
+use base64::encode;
+use serde::Deserialize;
+use serde_json::Value;
+use stacks::net::http::{HttpRequestContents, HttpResponsePayload};
+use stacks::net::httpcore::{send_http_request, StacksHttpRequest};
+use stacks::types::net::PeerHost;
+use url::Url;
+
+#[cfg(test)]
+mod tests;
+
+/// The JSON-RPC protocol version used in all requests.
+/// Latest specification is `2.0`
+const RCP_VERSION: &str = "2.0";
+
+/// Represents a JSON-RPC request payload sent to the server.
+#[derive(Serialize)]
+struct JsonRpcRequest {
+    /// JSON-RPC protocol version.
+    jsonrpc: String,
+    /// Unique identifier for the request.
+    id: String,
+    /// Name of the RPC method to invoke.
+    method: String,
+    /// Parameters to be passed to the RPC method.
+    params: serde_json::Value,
+}
+
+/// Represents a JSON-RPC response payload received from the server.
+#[derive(Deserialize, Debug)]
+struct JsonRpcResponse<T> {
+    /// ID matching the original request.
+    id: String,
+    /// Result returned from the RPC method, if successful.
+    result: Option<T>,
+    /// Error object returned by the RPC server, if the call failed.
+    error: Option<Value>,
+}
+
+/// Represents a JSON-RPC error encountered during a transport operation.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum RpcError {
```

Comment:
Unless the compiler tells you it's impossible, please wrap the inner error type directly, instead of converting it to a `String`.